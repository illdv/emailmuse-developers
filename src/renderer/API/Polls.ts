import { AxiosWrapper } from 'src/renderer/API/AxiosWrapper';
import { AxiosPromise } from 'axios';
import {
  ICreatePollsRequest,
  ICreatePollsResponse, ISavePollRequest,
  IUpdatePollsRequest,
} from 'src/renderer/component/Profile/Polls/flux/interfase';

/// Admin API
function getPolls(): AxiosPromise<ICreatePollsResponse> {
  return AxiosWrapper.get('/polls');
}

function updatePoll(pollId: string, request: IUpdatePollsRequest): any {
  return AxiosWrapper.put(`/polls/${pollId}`, request);
}

function deletePoll(pollId: string): any {
  return AxiosWrapper.deleteResponse2(`/polls/${pollId}`);
}

function createPolls(request: ICreatePollsRequest): any {
  return AxiosWrapper.post('/polls', request);
}

// User API
// Default value need for standard poll
function getPoll(pollId = '1'): any {
  return AxiosWrapper.get(`/polls/1`);
}

function savePoll(request: ISavePollRequest): any {
  return AxiosWrapper.post(`/answers`, request);
}

export const PollsAPI = {
  getPolls,
  savePoll,
  getPoll,
  createPolls,
  updatePoll,
  deletePoll,
};
