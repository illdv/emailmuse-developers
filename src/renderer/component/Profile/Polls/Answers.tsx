import * as React from 'react';
import { Component } from 'react';
import { IAnswer } from 'src/renderer/component/Profile/Polls/flux/interfase';
import {
  FormControl,
  FormControlLabel,
  Radio,
  RadioGroup,
  WithStyles,
  withStyles,
} from '@material-ui/core';

const styles = theme => ({});

export namespace AnswersSpace {
  export interface IProps {
    answers: IAnswer[];
    reply: any;
    classes?: any;
    answered: string;
  }

  export interface IState {}
}

export class Answers extends Component<
  AnswersSpace.IProps & WithStyles<any>,
  AnswersSpace.IState
> {
  state: AnswersSpace.IState = {};

  handleChange = event => {
    const answer = event.target.value;
    this.props.reply(answer);
  };

  render() {
    const { classes, answers, answered } = this.props;
    return (
      <FormControl required className={classes.formControl}>
        <RadioGroup
          name='pollAnswers'
          className={classes.group}
          value={answered}
        >
          {answers &&
            answers.map(answer => (
              <FormControlLabel
                key={answer.id}
                value={String(answer.id)}
                control={<Radio />}
                label={answer.body}
                onChange={this.handleChange}
                color='primary'
              />
            ))}
        </RadioGroup>
      </FormControl>
    );
  }
}

export default withStyles(styles)(Answers);
