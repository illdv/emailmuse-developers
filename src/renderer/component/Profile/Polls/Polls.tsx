import * as React from 'react';
import { Component } from 'react';
import { IGlobalState } from 'src/renderer/flux/rootReducers';
import { connect, Dispatch } from 'react-redux';
import { IPollsActions, PollsActions } from 'src/renderer/component/Profile/Polls/flux/actions';
import { bindModuleAction } from 'src/renderer/flux/saga/utils';
import { IPoll, IQuestion } from 'src/renderer/component/Profile/Polls/flux/interfase';
import { Grid, Paper, Typography, withStyles, } from '@material-ui/core';
import InCenter from 'src/renderer/common/InCenter';
import Answers from 'src/renderer/component/Profile/Polls/Answers';
import { Loading } from 'src/renderer/common/Loading';

const styles = theme => ({
  root: {
    height: '100%',
  },
  paper: {
    width: 630,
    height: 500,
    paddingLeft: 40,
    paddingRight: 40,
    paddingTop: 48,
    paddingBottom: 26,
  },
  question: {
    'width': '100%',
    'text-align': 'center',
  },
  center: {
    display: 'flex',
    justifyContent: 'center',
    alignItems: 'center',
  },
});

export namespace PollsSpace {
  export interface IState {
    answer: string;
  }

  export interface IProps {
    pallsActions: IPollsActions;
    polls: IPoll;
    currentQuestion: IQuestion;
    classes?: any;
  }
}

export class Polls extends Component<PollsSpace.IProps, PollsSpace.IState> {
  state: PollsSpace.IState = {
    answer: null,
  };

  next = (answer: string) => {
    this.setState({ answer });
    setTimeout(() => {
      this.props.pallsActions.nextQuestion.REQUEST({ answer });
    }, 500);
  }

  render() {
    const { classes } = this.props;
    const question = this.props.currentQuestion;
    if (question) {
      return (
        <InCenter>
          <Paper className={classes.paper}>
            <Grid item xs={12} className={classes.root}>
              <Grid
                className={classes.root}
                container
                spacing={16}
                alignItems={'center'}
                justify={'center'}
                direction={'column'}
              >
                <Grid item>
                  <Typography variant='title' gutterBottom>
                    {question.title}
                  </Typography>
                  <Typography variant='subheading' gutterBottom>
                    {question.description}
                  </Typography>
                </Grid>
                <Grid>
                  <Answers answers={question.answers} reply={this.next} answered={this.state.answer}/>
                </Grid>
              </Grid>
            </Grid>
          </Paper>
        </InCenter>
      );
    }
    return <Loading/>;
  }
}

const mapStateToProps = (state: IGlobalState) => ({
  polls: state.polls.poll,
  currentQuestion: state.polls.currentQuestion,
});

const mapDispatchToProps = (dispatch: Dispatch<any>) => ({
  pallsActions: bindModuleAction(PollsActions, dispatch),
});

export default withStyles(styles)
(connect(mapStateToProps, mapDispatchToProps)(Polls));
