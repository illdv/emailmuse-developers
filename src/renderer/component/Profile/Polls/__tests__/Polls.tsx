import * as React from 'react';
import { shallow } from 'enzyme';
import { Polls } from '../Polls';

describe('Polls', () => {
  const render = shallow(<Polls/>);
  test('Render UI', () => {
    expect(render).toMatchSnapshot();
  });
});
