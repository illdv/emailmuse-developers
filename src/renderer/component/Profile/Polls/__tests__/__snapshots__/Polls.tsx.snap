// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Polls Render UI 1`] = `
<h1>
  Hello Polls
</h1>
`;
